pkg/cloud/scope/rosacontrolplane.go

```diff
@@ -201,3 +202,18 @@ func (s *ROSAControlPlaneScope) PatchObject() error {
 func (s *ROSAControlPlaneScope) Close() error {
 	return s.PatchObject()
 }
+
+// GetMachinePools belong to the controlPlane managed cluster.
+func (s *ROSAControlPlaneScope) GetMachinePools(ctx context.Context) ([]expclusterv1.MachinePool, error) {
```

Comment:
This seems generic--is this defined elsewhere? cluster-api utils, maybe?

Reply:
I moved the func to util pkg so it can be re-used with any other controlPlane if needed